google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/spi/v1/MessageDispatcher.java

```diff
@@ -57,6 +61,9 @@
   @VisibleForTesting static final Duration PENDING_ACKS_SEND_DELAY = Duration.millis(100);
   private static final int MAX_ACK_DEADLINE_EXTENSION_SECS = 10 * 60; // 10m
 
+  private static final ScheduledExecutorService alarmsExecutor =
```

Comment:
I think this will block the JVM from exiting. [`Thread` class](https://docs.oracle.com/javase/7/docs/api/java/lang/Thread.html) documents that unless `Runtime.exit` is called, the JVM keeps running if there is at least one non-daemon thread. [`Executors` class](https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/Executors.html#defaultThreadFactory()) in turn documents that it creates non-daemon threads by default.

Could you test this by running a CLI application, maybe `CreateSubscriptionAndConsumeMessages.java`? I think unit tests call `Runtime.exit(0)` when all tests pass so it kills everything and we don't see the test hang.